src/pages/Notifications/Form/useSaveBehaviorGroup.tsx

```diff
@@ -0,0 +1,215 @@
+};
+
+const actionsNeedsUpdate = (
+    original: Partial<BehaviorGroup>['actions'] | undefined,
+    updated: (BehaviorGroup | NewBehaviorGroup)['actions']
+) => {
+    return original?.length !== updated?.length || !isEqual(actionsToIdList(original), actionsToIdList(updated));
+};
+
+export enum SaveBehaviorGroupResult {
+    CREATE,
+    UPDATE
+}
+
+type SaveActionsResponse = {
+    operation: SaveBehaviorGroupResult,
+    status: boolean;
+}
+
+const saveActions = async (
+    behaviorGroupId: UUID | undefined,
+    original: Partial<BehaviorGroup>['actions'] | undefined,
+    updated: (BehaviorGroup | NewBehaviorGroup)['actions'],
+    save: ReturnType<typeof useUpdateBehaviorGroupActionsMutation>['mutate'],
+    query: ReturnType<typeof useClient>['query'],
+    setFetchingIntegrations: Dispatch<SetStateAction<boolean>>
+): Promise<SaveActionsResponse> => {
+    if (actionsNeedsUpdate(original, updated)) {
+        // Determine what system Integrations we need to fetch
+        const toFetch: ReadonlyArray<SystemProperties> = uniqWith(
+            ([] as Array<SystemProperties>)
+            .concat(...updated.filter(isActionNotify)
+            .map(action => produce(action, draft => {
+                draft.recipient = draft.recipient.filter(r => !r.integrationId);
+            }))
+            .map(action => toSystemProperties(action))),
+            isEqual
+        );
+
+        if (toFetch.find(props => props.type !== NotificationType.EMAIL_SUBSCRIPTION)) {
+            throw new Error('Only email subscriptions are created when assigning behavior groups');
+        }
+
+        if (toFetch.length > 0) {
+            setFetchingIntegrations(true);
+        }
+
+        const response = await Promise.all(
+            toFetch.map(systemProps => query(getDefaultSystemEndpointAction(systemProps))
+            .then(result => result.payload?.type === 'Endpoint' ? result.payload.value.id : undefined)
+            )
+        ).then(newIds => {
+            if (newIds.includes(undefined)) {
+                throw new Error('Unexpected ids were returned when querying for system endpoints');
+            }
+
+            // We want to preserve the order
+            const remainingIds = [ ... newIds ] as UUID[];
+            const endpointsToAdd = actionsToIdList(updated, remainingIds);
+
+            return save({
+                behaviorGroupId: behaviorGroupId as UUID,
+                endpointIds: endpointsToAdd
+            });
+        });
+
+        if (response.payload?.status === 200) {
+            if (behaviorGroupId === undefined) {
+                return {
+                    operation: SaveBehaviorGroupResult.CREATE,
+                    status: true
+                };
+            } else {
+                return {
+                    operation: SaveBehaviorGroupResult.UPDATE,
+                    status: true
+                };
+            }
+        } else if (behaviorGroupId === undefined) {
```

Comment:
Why is this in here back to back? See lines below (@ 169)

Reply:
nvm, I see the return status is different.

Reply:
It's a bit difficult to see with the `if` nesting, but line 169 is outside of the `if actionsNeedsUpdate(original, updated)` at line 105.

So line 169 deals with the case when we didn't need to update, but are returning an OK response anyway.